### ISSUE TITLE
False positive for F823 undefined-local when redefining builtin with partial

### ISSUE THREAD

**Issue by Hnasar** [charliermarsh/ruff#4477]
Using ruff 0.0.267 If you have a function which redefines a builtin, then there's a false positive error for undefined-local

Repro:
```py
# foo.py
import functools,sys

def main():
    print = functools.partial(print, file=sys.stderr)
    print(3)

main()
```

```
ruff --select=F foo.py 
foo.py:4:31: F823 Local variable `print` referenced before assignment
```
